### PR TITLE
Add MCG spec section to the storage cluster CRD

### PR DESCRIPTION
**Pull request by nb-ohad**
And reconcile it as part for the noobaa system reconciler 
Currently focusing on exposing noobaa endpoint specs via the `storageclsuter` CRD

Signed-off-by: Ohad Mitrani omitrani@redhat.com

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,7 @@ $(call add-crd-gen,ocsv1,./pkg/apis/ocs/v1,./deploy/crds,./deploy/crds)
 	shellcheck-test \
 	gofmt \
 	golint \
-	govet \
+	gitgovet \
```
Comment (jarrpa) [inline Makefile:43]: Isn't the target still named `govet`? At least it is in this version of the file.

Comment (nb-ohad) [reply Makefile:43]: Yes, you are right, I guess I just wrote `git` while the focus was on the editor.
I will revert and push

Comment (nb-ohad) [reply Makefile]: done

**Pull request merged**